dali/operators/geometry/affine_transforms/transform_shear.cc

```diff
@@ -118,42 +127,49 @@ class TransformShearCPU
     assert(matrices.size() == static_cast<int>(shear_.size()));
     for (int i = 0; i < matrices.size(); i++) {
       auto &mat = matrices[i];
-      const mat3x2 &shear_factors = *reinterpret_cast<const mat3x2*>(shear_[i].data());
-      mat = shear(shear_factors);
+      if (shear_.IsDefined()) {
+        const mat3x2 &shear_factors = as_mat<3, 2>(shear_[i]);
+        mat = shear(shear_factors);
+      } else {
+        assert(angles_.IsDefined());
+        vec<6> shear_factors;
+        for (int j = 0; j < 6; j++)
+          shear_factors[j] = std::tan(deg2rad(angles_[i].data[j]));
+        mat = shear(*reinterpret_cast<mat3x2*>(&shear_factors));
+      }
       if (center_.IsDefined()) {
-        const vec3 &center = *reinterpret_cast<const vec3*>(center_[i].data());
+        const vec3 &center = as_vec<3>(center_[i]);
         mat.set_col(ndim, cat(sub<ndim, ndim>(mat) * -center + center, 1.0f));
       }
     }
   }
 
   void ProcessArgs(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
-    int repeat = IsConstantTransform() ? 0 : nsamples_;
+    auto shape_from_size =
+      [this](int64_t size) {
+        ndim_ = sqrt(size);
```

Comment:
I think that's wrong.
```suggestion
        ndim_ = std::ceil(sqrt(size));
```
and then use `(ndim_ - 1) * ndim_`